taskcluster/scripts/toolchain/repack-gradle.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -ex
+
+. "$(dirname $0)/directories.sh"
+
+pushd $PROJECT_DIR
+# No need to build anything to fetch gradle.
+./gradlew tasks
+popd
+
+tar cf - -C "$HOME/.gradle/" --transform='s,^\./,gradle/,' '.' \
```

Comment:
I ended up caching the full gradle home. I looked into it and it doesn't have any files tied to `assembleDebug`. It just contains what gradle generated to be able to run. I first used `~/.gradle/wrapper/dists` which just contains the jar files of the downloaded gradle. It output an archive that is half the size of the full home. That said, gradle generates these files anyway so I think it's okay to the full home. Moreover, we don't depend on how gradle stores its files.

Reply:
I went looking at the mozilla-central code for this, and it looks like it just finds the cached upstream distribution zip, and repacks that. And then the build directly calls gradle from there, rather than using the wrapper. I'm curious how much work that would be to do here, so that the automation here does not diverge further from m-c.